src/entities/MovieForm/ui/MovieForm.jsx

```diff
@@ -1,106 +1,176 @@
-import React, { useState } from "react";
+import React from "react";
+import { useOutletContext } from "react-router";
 import "./MovieForm.css";
 import { genresList } from "../../../entities/SelectTabs/selectTabsMock";
 import { getMovie } from "../lib/getMovie";
+import { useForm, Controller } from "react-hook-form";
+import { MAIN_API } from "../../../app/config";
 
-export const MovieForm = ({ initialMovie, onFormSubmit }) => {
-  const [movie, setMovie] = useState(getMovie(initialMovie));
+export const MovieForm = ({ initialMovie }) => {
+  const { movie } = useOutletContext() ?? {initialMovie};
+ 
+  const { handleSubmit, control, reset } = useForm({
+    defaultValues: getMovie(movie),
+  });
 
-  const handleFieldChange = (event) => {
-    setMovie((prevMovie) => ({
-      ...prevMovie,
-      [event.target.name]: event.target.value,
-    }));
+  const onFormSubmit = async (data) => {
+    let requestOptions = {
+      method: "POST",
+      headers: {
```

Comment:
in general, it should be managed in one place not every time for each request, but for now it's okay, let's keep